File: LeetCode/Python Solutions/Reduction Operations to Make the Array Elements Equal/reduce.py
from typing import List

class Solution:
    def reductionOperations(self, nums: List[int]) -> int:
        nums.sort()  # Sort in non-decreasing order
        operations = 0
        distinct_counts = 0

        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                distinct_counts += 1
            operations += distinct_counts

        return operations

File: LeetCode/Java Solutions/Reduction Operations to Make the Array Elements Equal/reduce.java
import java.util.Arrays;

class Solution {
    public int reductionOperations(int[] nums) {
        Arrays.sort(nums); // Sort in non-decreasing order
        int operations = 0;
        int distinctCounts = 0;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i - 1]) {
                distinctCounts++;
            }
            operations += distinctCounts;
        }

        return operations;
    }
}
